UKHomeOffice/sgmr-service
# NMSW-34 Add autocomplete to port fields

## AC

Replace the existing autocomplete component with the AlphaGov version that we use on nmsw

Refactor port field component to use new autocomplete component
Update arrival and departure forms to work with new port field

----

## To test

- create a voyage
- fill in departure details
- start typing a departure port
- > you should see a select list
- select a port
- > port should be shown in the field
- refresh the page
- > port should persist
- click the link for the other port field
- type in the field
- refresh the page
- > port should persist what you typed in that field
- type in the autocomplete field
- select a port
- > other port field should hide
- save and continue
- leave voyage flow
- return to voyage from the dashboard
- click 'change' for departure on CYA page
- > port should be prefilled
- select location not in list
- type in free text field
- save and continue
- leave voyage flow
- return to voyage
- click 'change' for departure on CYA page
- > port should have your free text value
- repeat for arrival
- complete voyage
- submit voyage
- > voyage should submit

----

## Notes
